invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileUploader.js

```diff
@@ -79,6 +81,9 @@ export const FileUploaderComponent = ({
       );
       const maxFileStorageReached = filesSize + acceptedFilesSize > quota.maxStorage;
 
+      const emptyFiles = acceptedFiles.filter((file) => file.size === 0);
+      const nonEmptyFiles = acceptedFiles.filter((file) => file.size > 0);
```

Comment:
You've removed the arrays emptyFiles and duplicateFiles and replaced them with boolean flags like hasEmptyFiles and hasDuplicatedFiles. But, the arrays emptyFiles and duplicateFiles are necessary because they're used to display the file names in the warning messages. Without these arrays, the app cannot list the specific file names that are empty or duplicated, this is why I added them, just let me know if there's a better way to do it.